pkg/client/userd/trafficmgr/install.go

```diff
@@ -210,20 +215,13 @@ var agentNotFound = errors.New("no such agent")
 
 func (ki *installer) getSvcForInjectedPod(
 	c context.Context,
-	name,
-	namespace,
-	svcName,
-	portNameOrNumber string,
+	svc *core.Service,
+	name, namespace string,
 	podTemplate *core.PodTemplateSpec,
 	obj k8sapi.Object,
 ) (*core.Service, error) {
```

Comment:
The method should only return `error` now. There's no point in always returning the service that was passed in because the caller already has it. Returning it here give the impression that the method might change it.